### DISCUSSION TITLE
Error in relations

### DISCUSSION THREAD

**Discussion by alukardua1** [rappasoft/laravel-livewire-tables#1019] [Feature Requests]
I output relations and somehow the code works strangely
DataTable
`protected $model = Task::class;
	
	public function builder(): Builder
	{
		if(in_array(Auth::user()->id, config('myConfig.authorTask'))){
			return $this->model::with(['getAuthor:id,first_name', 'getResponsible:id,first_name'])->where('author_id', Auth::user()->id);
		}
		//dd($this->model::with(['getAuthor:id,first_name', 'getUser:id,first_name'])->get());
		return $this->model::with(['getAuthor:id,first_name', 'getUser:id,first_name'])->where('responsible_id', Auth::user()->id);
	}
	
	public function columns(): array
	{
		return [
			Column::make("Id", "id")->sortable(),
			Column::make('number_task', 'number_task')->sortable(),
			Column::make("task_date", "task_date")->format(function($value, $row) {
				echo $this->formatDate($row->task_date);
			})->sortable(),
			Column::make('task_date_to', 'task_date_to')->format(function($value, $row) {
				echo $this->formatDate($row->task_date_to);
			}),
			Column::make('end_task', 'end_task')->format(function($value, $row) {
				echo $this->formatDate($row->end_task);
			}),
			Column::make('author', 'getAuthor.first_name'),
			Column::make('responsible', 'getUser.first_name'),
			Column::make('name_task', 'name_task'),
			Column::make('description_task', 'description_task'),
			BooleanColumn::make('read', 'read'),
		];
	}`
	Model
	`public string $cachePrefix = 'task_';
	
	protected $fillable = [
		'task_date',
		'task_date_to',
		'author_id',
		'number_task',
		'name_task',
		'description_task',
		'responsible_id',
		'read',
		'end_task',
	];
	
	public function getAuthor(): hasOne
	{
		return $this->hasOne(User::class, 'id', 'author_id');
	}
	public function getUser()
	{
		return $this->hasOne(User::class, 'id', 'responsible_id');
	}`
	and in the table it outputs the same thing in both columns of getAuthor

**Comment by lrljoe** [comment]
Couple of comments on this one in relation to your model.

1. Make sure you're using belongsTo() rather than hasOne(), so that the inverse of the relationship is correct.
2. Don't allow foreign keys to be fillable, I know its convenient, but its a relatively common attack vector when using Livewire.

In relation to your table, if you use the format function to display the value of the field instead, does it work correctly?

**Comment by alukardua1** [comment]
I solved this problem instead of 
`Column::make('responsible', 'getUser.first_name'),`
used the 
`Сolumn::make(__('app.responsible'), 'responsible_id')->format(function($value, $row) {
				echo $row->getUser->user_name;
			}),`

**Comment by lrljoe** [comment]
You definitely absolutely do it that way, but you may see an increased number of queries against your DB, which may impact scalability in the future.  You'll want to eager load where you can to reduce db impact at scale.

Run debugbar against it and see what it shows to make sure you're comfortable.

If you intend to sort those fields, then I've made other posts on here discussing how to do that when using formatted output on this style

If you want me to cast my eye over it properly then please make a basic repo with the necessary models and table and I'll look further over the coming week and let you know what best practice looks like.
Or I'm happy to jump on a screen share with enough notice.

**Comment by alukardua1** [comment]
In principle, I use the module "rennokki/laravel-eloquent-query-cache" and the number of queries to the database is not so critical
